qiskit_dynamics/signals/signals.py

```diff
@@ -303,38 +290,27 @@ def __init__(
         """
         self._dt = dt
 
-        samples = Array(samples)
+        samples = unp.asarray(samples)
 
         if len(samples) == 0:
-            zero_pad = np.array([0])
+            zero_pad = unp.asarray([0])
         else:
-            zero_pad = np.expand_dims(np.zeros_like(Array(samples[0])), 0)
-        self._padded_samples = np.append(samples, zero_pad, axis=0)
+            zero_pad = unp.expand_dims(unp.zeros_like(unp.asarray(samples[0])), 0)
```

Comment:
can you try changing this to:
```suggestion
            zero_pad = unp.expand_dims(unp.zeros_like(samples[0]), 0)
```
samples is already being passed through `unp.asarray` earlier in the method. The previous code had to wrap this in `Array` again as indexing would remove `Array` wrapping, but that shouldn't be necessary here.